src/instructions/interrupts.rs

```diff
@@ -130,26 +121,30 @@ where
 #[inline]
 pub fn enable_and_hlt() {
     unsafe {
-        #[cfg(feature = "inline_asm")]
         asm!("sti; hlt", options(nomem, nostack));
-
-        #[cfg(not(feature = "inline_asm"))]
-        crate::asm::x86_64_asm_interrupt_enable_and_hlt();
     }
 }
 
 /// Cause a breakpoint exception by invoking the `int3` instruction.
 #[inline]
 pub fn int3() {
     unsafe {
-        #[cfg(feature = "inline_asm")]
         asm!("int3", options(nomem, nostack));
-
-        #[cfg(not(feature = "inline_asm"))]
-        crate::asm::x86_64_asm_int3();
     }
 }
 
+/// Generate a software interrupt by invoking the `int` instruction.
+///
+/// This currently needs to be a macro because the `int` argument needs to be an
+/// immediate. This macro will be replaced by a generic function when support for
+/// const generics is implemented in Rust.
+#[macro_export]
+macro_rules! software_interrupt {
+    ($x:expr) => {{
+        asm!("int {id}", id = const $x, options(nomem, nostack));
+    }};
+}
+
```

Comment:
I don't think keeping this macro around actually lets us call `software_interrupt!(0x42)` on stable. At least when I try to do this, [I still get an error about `asm_const`](https://rust.godbolt.org/z/dx38a3sbT). I don't think 0.14 has a stable way to do this, so we don't lose anything by only having the `software_interrupt` function.

In fact, having the function is slightly nicer, as the x86_64 crate can enable the feature for the user, rather than the user having to manually deal with it.

Reply:
Oh yeah, you're right. 

You can get it to work if you're willing to commit [some macro crimes](https://rust.godbolt.org/z/Tne8Tzrsb) ;)

Reply:
That marco approach is interesting, but I think it might lead to confusing behavior. For example, this works
```rust
software_interrupt!(2*3 + 7 - 2*3);
```
but this doesn't:
```rust
const N: u8 = 5;
software_interrupt!(N);
```
nor does this:
```rust
software_interrupt!(5u8);
```

So having this might be confusing to users.

Reply:
I'm fine with removing the macro, given that it's apparently unusable in its current form on stable.

If there is demand for it, we can still add a stable-compatible macro at a later time, but I would wait until it is requested.